types/react-table/index.d.ts

```diff
@@ -864,7 +864,7 @@ export function defaultOrderByFn<D extends object = {}>(
 export function defaultGroupByFn<D extends object = {}>(
     rows: Array<Row<D>>,
     columnId: IdType<D>,
-): Record<string, Row<D>>;
+): Record<string, Array<Row<D>>>;
```

Comment:
By default, a reduce function will return a single value, in this case, an Object (see https://github.com/tannerlinsley/react-table/blob/99d81d9a967c0c628706ef79309118378f0ad7ee/src/plugin-hooks/useGroupBy.js#L419). Therefore, this wouldn't be an array, and this change would theoretically break stuff. @ggascoigne Please correct me if I'm reading the code wrong.

Reply:
A `Record` is basically an alias for an object though. I'm not suggesting changing the outer container for the value. 

`Record<string, Array<Row<D>>>` is roughly equalivent at runtime to:

```
{
  "someId": [] // list of rows
}

```

Reply:
The current definion is roughly at runtime:
```
{
  "someId" : // some row object
}
```

Reply:
The current default implementation of this type literatally checks that all unitialised values within the record gets initilaised as an Array and then pushes into that array.

Reply:
There's the difference: 
` // list of rows` or ` // some row object`. I believe the reduce function returns a single row object, not a list of rows. I could certainly be wrong, but I think that's the difference.

Reply:
Admittedly, I'll have to dig into it further to see the actual execution versus what my preconceived thoughts are.

Reply:
Here is an example where I have directly lifted the default implementation:
https://jsfiddle.net/ve4qcapr/

Reply:
OK, looking through it, I realize where my confusion was. It is returning a single object (which is the Record), which consists of the key (string) and the value (Array) (which you stated).

I'm guessing there aren't a lot of people using groupBy, or this would have been found sooner. :) Good catch @m-rutter!

Reply:
and here is a typescript playground example to help:
https://www.typescriptlang.org/play/?ssl=21&ssc=1&pln=21&pc=20#code/JYOwLgpgTgZghgYwgAgEoHsDuAeAKsgXmQFcQATCGUCMgPmQG8AoZZANzgBtiIBnALjQQE6KGWy8wUUAHMANMjggAnrQDcLZGAAWsgPyDcTAL5MmMUgjDB0IZBXjFOYAOJR0xAA4AhZQDEQPFoACndMATQsIIBtAF0FEW4AWxAASTJBSWkQGQBKQVRhUXEs2QUAQSgoOGVsDBxcWibGTSgIMGIoOzDeADo2smIkOqKxCSky5Erq2vqgppDgzza2BTCFYFzCemZWVgB6feRcAHkAERPBAFlgGW0wZEwlB7B0ZGAkz04IJIhwRWQAGsIMpkLxoMAuMAAF7QZDaaAoXjoTQHI7AKoQGROOBQZCJYgpdhcHi8fFKMHWTicZAAIxQMncXho7xgyBAEBoND0qPxtkkyDavAA0iDCMgAAYAEgYYV6HG4fGiBJS6VixglvOWEDY0SFouUsXF0xqvWAvBNyiWKz1fANsS2emQ2t1+pBRsEcS1Nrdht6nmIvG0oSwuV5bQ6XWdK00xgUDGMYdMTBEIAFPQKWDi4uiDAVpMEDAARMjfgBhdCcIuCACMxmMsTMqfTfCcDyIDjgbbcHh8-hAIfCChL6HLlaLYZT-MrEF6nHQMlCrecuSAA

Reply:
Yeah, I don't think anyone is using it, or someone would have reported this by now.